test/powershell/Host/Logging.Tests.ps1

```diff
@@ -0,0 +1,232 @@
+        [bool] $persistenceEnabled = $false
+
+        if ($IsSupportedEnvironment)
+        {
+            # Check the current state.
+            $persistenceEnabled  = (Get-OSLogPersistence).Enabled
+            if (!$persistenceEnabled)
+            {
+                # enable powershell log persistence to support exporting log entries
+                # for each test
+                 Set-OsLogPersistence -Enable
+            }
+        }
+        [string] $powershell = Join-Path -Path $PSHome -ChildPath 'pwsh'
+    }
+
+    BeforeEach {
+        if ($IsSupportedEnvironment)
+        {
+            # generate a unique log application id
+            [string] $logId = [Guid]::NewGuid().ToString('N')
+
+            # Generate a working directory and content file for Export-OSLog
+            [string] $workingDirectory = Join-Path -Path $TestDrive -ChildPath $logId
+            $null = New-Item -Path $workingDirectory -ItemType Directory -ErrorAction Stop
+
+            [string] $contentFile = Join-Path -Path $workingDirectory -ChildPath ('pwsh.log.txt')
+            # get log items after current time.
+            [DateTime] $after = [DateTime]::Now
+        }
+    }
+
+    AfterAll {
+        if ($IsSupportedEnvironment -and !$persistenceEnabled)
+        {
+            # disable persistence if it wasn't enabled
+             Set-OsLogPersistence -Disable
+        }
+    }
+
+    It 'Verifies basic logging with no customizations' -Skip:(!$IsSupportedEnvironment) {
+        $configFile = WriteLogSettings -LogId $logId
+        & $powershell -NoProfile -SettingsFile $configFile -Command '$env:PSModulePath | out-null'
+
+        Export-PSOsLog -After $after -Verbose | Set-Content -Path $contentFile
+        $items = Get-PSOsLog -Path $contentFile -Id $logId -After $after -TotalCount 3 -Verbose
+
+        $items | Should Not Be $null
+        $items.Length | Should BeGreaterThan 1
+        $items[0].EventId | Should Be 'Perftrack_ConsoleStartupStart:PowershellConsoleStartup.WinStart.Informational'
+        $items[1].EventId | Should Be 'Perftrack_ConsoleStartupStop:PowershellConsoleStartup.WinStop.Informational'
+        # if there are more items than expected...
+        if ($items.Length -gt 2)
+        {
+            # Force reporting of the first unexpected item to help diagnosis
+            $items[2] | Should be $null
+        }
+    }
+
+    It 'Verifies logging level filtering works' -Skip:(!$IsSupportedEnvironment) {
+        $configFile = WriteLogSettings -LogId $logId -LogLevel Warning
+        & $powershell -NoProfile -SettingsFile $configFile -Command '$env:PSModulePath | out-null'
+
+        Export-PSOsLog -After $after -Verbose | Set-Content -Path $contentFile
+        # by default, powershell startup should only log informational events are logged.
+        # With Level = Warning, nothing should
+        $items = Get-PSOsLog -Path $contentFile -Id $logId -After $after -TotalCount 3
+        $items | Should Be $null
```

Comment:
`Should -BeNullOrEmpty` or `Should -Be $null`

Reply:
Fixed